From b9cedf712cc1298347fead9b014495e3c0591b07 Mon Sep 17 00:00:00 2001
From: OpenEmbedded <oe.patch@oe>
Date: Wed, 26 Apr 2023 17:01:13 +0000
Subject: [PATCH] commit	95aab0fd83619408995720ce53d7a74790580220 author
 liberato@chromium.org <liberato@chromium.org>	Thu Jul 08 02:01:22 2021
 committer	liberato@chromium.org <liberato@chromium.org>	Thu Jul 08
 02:01:22 2021 tree	ac725b5e2c548c8142aa7096d8184d87d3876a49 parent
 e073b7a22e4993e0a7cab80a42a21524e5349f95

Add av_stream_get_first_dts for Chromium

---
 libavformat/avformat.h | 4 ++++
 libavformat/utils.c    | 7 +++++++
 2 files changed, 11 insertions(+)

diff --git a/libavformat/avformat.h b/libavformat/avformat.h
index 6ce367e..cba3f5e 100644
--- a/libavformat/avformat.h
+++ b/libavformat/avformat.h
@@ -1115,6 +1115,10 @@ struct AVCodecParserContext *av_stream_get_parser(const AVStream *s);
  */
 int64_t    av_stream_get_end_pts(const AVStream *st);
 
+// Chromium: We use the internal field first_dts vvv
+int64_t    av_stream_get_first_dts(const AVStream *st);
+// Chromium: We use the internal field first_dts ^^^
+
 #define AV_PROGRAM_RUNNING 1
 
 /**
diff --git a/libavformat/utils.c b/libavformat/utils.c
index cee86ae..6a01413 100644
--- a/libavformat/utils.c
+++ b/libavformat/utils.c
@@ -150,6 +150,13 @@ const AVCodec *ff_find_decoder(AVFormatContext *s, const AVStream *st,
     return avcodec_find_decoder(codec_id);
 }
 
+// Chromium: We use the internal field first_dts vvv
+int64_t av_stream_get_first_dts(const AVStream *st)
+{
+  return cffstream(st)->first_dts;
+}
+// Chromium: We use the internal field first_dts ^^^
+
 /* an arbitrarily chosen "sane" max packet size -- 50M */
 #define SANE_CHUNK_SIZE (50000000)
 
